libs/etterlatte-regler/src/main/kotlin/regler/PeriodisertGrunnlag.kt

```diff
@@ -0,0 +1,21 @@
+package no.nav.etterlatte.libs.regler
+
+import java.time.LocalDate
+
+interface PeriodisertGrunnlag<G> {
+    fun finnAlleKnekkpunkter(): Set<LocalDate>
+
+    fun finnKnekkpunkterInnenforPeriode(periode: RegelPeriode): Set<LocalDate> {
+        return finnAlleKnekkpunkter().filter { knekkpunkt ->
+            knekkpunkt >= periode.fraDato && (periode.tilDato?.let { it >= knekkpunkt } ?: true)
+        }.toSet()
+    }
+
+    fun finnGrunnlagForPeriode(datoIPeriode: LocalDate): G
+}
+
+class KonstantGrunnlag<G>(private val konstantGrunnlag: G) : PeriodisertGrunnlag<G> {
```

Comment:
Hva betyr det egentlig at det er et Konstant grunnlag med periodiserte opplysninger? Jeg trodde det var to motsetninger.

Reply:
Jeg vet ikke om du har noen forslag til bedre navngivning, men for meg er det ikke noen motsetning med å ha noe som er konstant over perioder. F.eks. funksjonen `f(x) = 2` er en konstant funksjon, og uavhengig av hvilken `x` du putter inn er svaret `2`. Men det er likefullt en funksjon, og man kan se på den likt som andre funksjoner etc.